tests/vdir_test.py

```diff
@@ -86,3 +85,16 @@ def test_etag_sleep(tmpdir, sleep_time):
     new_etag = vdir.get_etag_from_file(fpath)
 
     assert old_etag != new_etag
+
+
+def test_get_href_from_uid():
+    uid = "V042MJ8B3SJNFXQOJL6P53OFMHJE8Z3VZWOU@pimutils.org"
+    href = vdir._generate_href(uid)
+    assert href == "4aa4284e48c5b195d48e3f9a8fd29bd574c66f47"
```

Comment:
could we call vdir._generate_href(uid) here twice and check if it is the same output? Because this is different to the behaviour before, isn't it?

Reply:
Yes good idea :) I modify my PR in this way

Reply:
Ok I rework the whole test